### PR TITLE
Adds IBERT to models exportable with ONNX

### PR DESCRIPTION
**Pull request by MaximovaIrina**
Adds IBERT to models exportable with ONNX

### CODE REVIEW

**Review by LysandreJik — COMMENTED** [review]
Nice!! Pinging @michaelbenayoun and @lewtun

File: src/transformers/models/ibert/configuration_ibert.py
Comment (lewtun) [inline src/transformers/models/ibert/configuration_ibert.py:132]: For encoder-based models like IBERT, I think it makes more sense to use `OnnxConfig` instead of `OnnxConfigWithPast`. The `OnnxConfigWithPast` is usually reserved for decoder-based and encoder-decoder models where having access to the past key values is important for text generation.

Is there a reason why you chose this particular config?

**Review by lewtun — COMMENTED** [review]
Thank you for this very clean contribution @MaximovaIrina !

I've left a minor remark, but otherwise this is looking really good. Could you please check that the "slow" tests pass for this model:

```
RUN_SLOW=1 pytest tests/test_onnx_v2.py -k "ibert" -rp
```

These tests aren't run until the PR is merged, so it's good to know in advance if there's any problems :)

File: src/transformers/models/ibert/configuration_ibert.py
Comment (MaximovaIrina) [reply src/transformers/models/ibert/configuration_ibert.py:132]: I followed the recommendations in this [tutorial](https://huggingface.co/docs/transformers/serialization#converting-an-onnx-model-using-the-transformersonnx-package). In particular, I was looking at this [patch](https://github.com/huggingface/transformers/pull/13049/commits/d097adcebd89a520f04352eb215a85916934204f).

File: src/transformers/models/ibert/configuration_ibert.py
Comment (michaelbenayoun) [inline src/transformers/models/ibert/configuration_ibert.py:143]: You do not need to override the `outputs` property as it is already taken care of by `OnnxConfig` for each of the tasks supported by IBERT (the outputs will depend on the task at hand, so the base class `outputs` property takes care of returning the proper outputs for a given task).

File: src/transformers/onnx/features.py
Comment (michaelbenayoun) [inline src/transformers/onnx/features.py:67]: Can you please add all the supported features for IBERT?
Those are:

- masked-lm
- sequence-classification
- ~~multiple-choice~~ (this one is not supported yet)
- token-classification
- question-answering

File: src/transformers/models/ibert/configuration_ibert.py
Comment (lewtun) [reply src/transformers/models/ibert/configuration_ibert.py:132]: Thanks for the information @MaximovaIrina - we're in the process of revamping the docs / tutorial, so hopefully it will be less confusing in future!

The patch you linked to is for MBART, which is an encoder-decoder model where having access to past key values is important. In your case with IBERT, I recommend looking at the [DistilBERT example](https://github.com/huggingface/transformers/blob/6b655cc63fd7e8ec120d9c38321a286d04767db0/src/transformers/models/distilbert/configuration_distilbert.py#L136) which inherits from `OnnxConfig`.

As @michaelbenayoun notes elsewhere, you can also use this as a guide for adding all the supported features.

**Comment by lewtun** [comment]
Hey @MaximovaIrina we recently switched all of our documentation from RST to MDX, so you can rebase on `master` to eliminate the conflict with `docs/source/serialization.rst`

File: docs/source/serialization.rst
Comment (lewtun) [inline docs/source/serialization.rst:2]: This file shouldn't be included (must be a Git problem with the switch from RST to MDX). Could you please delete it from the PR?

Comment (MaximovaIrina) [reply docs/source/serialization.rst:2]: As you say

Comment (lewtun) [reply docs/source/serialization.rst:2]: Hmm I still see the file in the PR - is it possible you only reverted the diff? If a normal `rm docs/source/serialization.rst` + git commit doesn't fix it, perhaps running 

```
git rm docs/source/serialization.rst
```

does?

Comment (MaximovaIrina) [reply docs/source/serialization.rst]: Evsen should be fine now

**Review by lewtun — APPROVED** [review]
Thank you very much for these last round of changes @MaximovaIrina !

Once the CI tests pass, this PR LGTM 🚀 - OK for you too @LysandreJik ?

**Comment by lewtun** [comment]
Hey @MaximovaIrina would you mind rebasing your branch on `master` and resolving the merge conflicts?

Gently pinging @LysandreJik for his blessing on this PR too :)

**Review by LysandreJik — APPROVED** [review]
Looks good to me! Thanks for your PR, @MaximovaIrina!

**Comment by LysandreJik** [comment]
Could you just run the code quality tool to ensure that the code quality passes? You can install them with the following, from the root of your clone:
```
pip install -e ".[quality]"
```
And then run them with:
```
make fixup
```

**Comment by lewtun** [comment]
Merging this since all the CI checks now pass and we have approval from a core maintainer. Thank you for your contribution @MaximovaIrina 🤗 !

**Pull request merged**
